drivers/i2c/Kconfig.dw

```diff
@@ -5,7 +5,7 @@ menuconfig I2C_DW
 	bool "Design Ware I2C support"
 	default y
 	depends on DT_HAS_SNPS_DESIGNWARE_I2C_ENABLED
-	select PINCTRL if DT_HAS_RASPBERRYPI_PICO_I2C_ENABLED
+	select PINCTRL if $(dt_compat_any_has_prop,$(DT_COMPAT_I2C_DW),pinctrl-0)
```

Comment:
The compatible is called snps,designware-i2c, so maybe you should check something like DT_COMPAT_SNPS_DESIGNWARE_I2C?

Reply:
This is an IP core that is used by multiple SoCs. Some of them do not have pinctrl, so we need to check.

Reply:
> This is an IP core that is used by multiple SoCs. Some of them do not have pinctrl, so we need to check.

Yeah, that's for sure. I've done the same for the DW SPI core. My only point is the specific compatible name, which is used to generate the macros you're trying to use.

Reply:
Ah, I got it. It was still a draft, not yet tested. Thank you.

Reply:
It worked fine with applying your point. Thank you.